substrate-node/pallets/pallet-tfgrid/src/node.rs

```diff
@@ -361,31 +361,31 @@ impl<T: Config> Pallet<T> {
         let twin_id = TwinIdByAccountID::<T>::get(account_id).ok_or(Error::<T>::TwinNotExists)?;
         let node = Nodes::<T>::get(node_id).ok_or(Error::<T>::NodeNotExists)?;
         let farm = Farms::<T>::get(node.farm_id).ok_or(Error::<T>::FarmNotExists)?;
+
+        // Make sure only the farmer that owns this node can change the power target
         ensure!(
             twin_id == farm.twin_id,
             Error::<T>::UnauthorizedToChangePowerTarget
         );
-        // Make sure only the farmer that owns this node can change the power target
+
+        // When power target is switched to Down, make sure there are no active contracts on node
         ensure!(
-            node.farm_id == farm.id,
-            Error::<T>::UnauthorizedToChangePowerTarget
+            power_target == Power::Down
```

Comment:
But this will prevent setting the node power target to up?

Instead, you should use if condition to check if node power target is down then inside it you can verify that node don't have active contacts.